### ISSUE TITLE
parse_file() expects some extra curly brackets if next value is not a list.

### ISSUE THREAD

**Issue by alexey-terentiev** [chimpler/pyhocon#56]
Config file test.conf:
```
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = bar
```
Code:
```
from pyhocon import ConfigFactory
config = ConfigFactory.parse_file("test.conf")
print(config['foo'])
```
Result:
```
Traceback (most recent call last):
  File "<Project_path>/hocon.py", line 2, in <module>
    config = ConfigFactory.parse_file("test.conf")
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 49, in parse_file
    return ConfigFactory.parse_string(content, os.path.dirname(filename), resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 88, in parse_string
    return ConfigParser().parse(content, basedir, resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 241, in parse
    config = config_expr.parseString(content, parseAll=True)[0]
  File "C:\Python27\lib\site-packages\pyparsing.py", line 1125, in parseString
    raise exc
pyparsing.ParseSyntaxException: Expected "{" (at char 105), (line:6, col:10)
```
Expected result: `bar`

**Comment by alexey-terentiev** [comment]
It works if I add brackets:
```
{
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = bar
}
```
-> `bar`
It works if `foo` is a list:
```
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = [ bar ]
```
-> `['bar']`

**Comment by alexey-terentiev** [comment]
It works if value is in double quotes:
```
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = "bar"
```
-> `bar`

**Comment by alexey-terentiev** [comment]
But it does not if it's in single quotes:
```
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = 'bar'
```
```
Traceback (most recent call last):
  File "<Project_path>/hocon.py", line 2, in <module>
    config = ConfigFactory.parse_file("test.conf")
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 49, in parse_file
    return ConfigFactory.parse_string(content, os.path.dirname(filename), resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 88, in parse_string
    return ConfigParser().parse(content, basedir, resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 241, in parse
    config = config_expr.parseString(content, parseAll=True)[0]
  File "C:\Python27\lib\site-packages\pyparsing.py", line 1125, in parseString
    raise exc
pyparsing.ParseException: Expected end of text (at char 98), (line:6, col:7)
```
Works with extra curly brackets:
```
{
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = 'bar'
}
```

**Comment by darthbear** [comment]
Hi @alexey-terentiev, thanks for the bug report. I tried on the latest version and I cannot reproduce the issue. Also thanks for the other bug reports, I'll release a new version with the fixes soon.

**Comment by alexey-terentiev** [comment]
I've just reproduced it with python 2.7.10 and pyhocon 0.3.15

**Comment by alexey-terentiev** [comment]
```
>C:\Python27\python.exe test.py
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    config = ConfigFactory.parse_file("test.conf")
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 49, in parse_file
    return ConfigFactory.parse_string(content, os.path.dirname(filename), resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 88, in parse_string
    return ConfigParser().parse(content, basedir, resolve)
  File "C:\Python27\lib\site-packages\pyhocon\config_parser.py", line 241, in parse
    config = config_expr.parseString(content, parseAll=True)[0]
  File "C:\Python27\lib\site-packages\pyparsing.py", line 1125, in parseString
    raise exc
pyparsing.ParseSyntaxException: Expected "{" (at char 101), (line:6, col:10)

>type test.py
from pyhocon import ConfigFactory
config = ConfigFactory.parse_file("test.conf")
print(config['foo'])

>type test.conf
app.heap_size = 128
app.java_opts = [
    -Xms${app.heap_size}m
    -Xmx${app.heap_size}m
]
foo = bar
```

**Comment by darthbear** [comment]
Thanks for getting back to me and for your detailed comments. I wonder if it's not because of `\r` characters that can be present at the end of lines `\r\n`. I created a new branch `dos-chars-fix` that might fix it. Do you mind attaching your file to this issue? Thanks!

**Comment by alexey-terentiev** [comment]
I don't mind but I can't: `Attaching documents requires write permission to this repository.`

**Comment by alexey-terentiev** [comment]
![image](https://cloud.githubusercontent.com/assets/15238370/10846729/d28c8e16-7f22-11e5-8891-cf49b506a301.png)

**Comment by darthbear** [comment]
Thanks @alexey-terentiev. I tried on my mac with the `master` branch and I got this error:
```
/Users/fdang/chimpler/pyhocon/pyhocon/config_tree.pyc in transform(self)
    332                         req_tok_type=first_tok_type.__name__,
    333                         line=lineno(self._loc, self._instring),
--> 334                         col=col(self._loc, self._instring)))
    335
    336         if first_tok_type is ConfigTree:

ConfigWrongTypeException: Token '' of type str (index 1) must be of type ConfigList (line: 5, col: 1)
```

A little different message but an error nonetheless. With the new branch `dos-chars-fix` it seems to be working fine. I'll see if I can a hold of a windows box to try it.

Thanks!

**Comment by alexey-terentiev** [comment]
Centos7.1, Python 2.7.5:
```
[osboxes@osboxes ~]$ python test.py 
Traceback (most recent call last):
  File "test.py", line 2, in <module>
    config = ConfigFactory.parse_file("test.conf")
  File "build/bdist.linux-x86_64/egg/pyhocon/config_parser.py", line 49, in parse_file
  File "build/bdist.linux-x86_64/egg/pyhocon/config_parser.py", line 88, in parse_string
  File "build/bdist.linux-x86_64/egg/pyhocon/config_parser.py", line 241, in parse
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 1115, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2378, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2483, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2624, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2739, in parseImpl
    loc, tmptokens = self.expr._parse( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2483, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2624, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2624, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2369, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2483, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 989, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 2369, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/usr/lib/python2.7/site-packages/pyparsing-2.0.3-py2.7.egg/pyparsing.py", line 995, in _parseNoCache
    tokens = self.postParse( instring, loc, tokens )
  File "build/bdist.linux-x86_64/egg/pyhocon/config_parser.py", line 388, in postParse
  File "build/bdist.linux-x86_64/egg/pyhocon/config_tree.py", line 334, in transform
pyhocon.exceptions.ConfigWrongTypeException: Token '' of type str (index 1) must be of type ConfigList (line: 5, col: 1)
[osboxes@osboxes ~]$ 
```

**Comment by darthbear** [comment]
Thanks @alexey-terentiev again. I actually found that parsing a file with no EOL on the last line resulted in the same exception that you described earlier. Wonder if that could be the problem. I pushed a fix to `dos-chars-fix`. Hope that fixes the issue :-). Thanks!

**Comment by alexey-terentiev** [comment]
@darthbear , seems like it's fixed.

**Comment by darthbear** [comment]
Thanks again for your help fixing this issue. It's now released `pyhocon==0.3.16`. Thanks!
